exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -491,259 +648,585 @@ func (e *Exchange) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Sub
 	if err := s.Validate(e.GetTradingRequirements()); err != nil {
 		return nil, err
 	}
-
-	fPair, err := e.FormatExchangeCurrency(s.Pair, s.AssetType)
+	var err error
+	s.Pair, err = e.FormatExchangeCurrency(s.Pair, s.AssetType)
 	if err != nil {
 		return nil, err
 	}
-	response, err := e.PlaceOrder(ctx,
-		fPair.String(),
-		s.Price,
-		s.Amount,
-		false,
-		s.Type == order.Market,
-		s.Side.IsLong())
-	if err != nil {
-		return nil, err
+	switch s.AssetType {
+	case asset.Spot:
+		var stpMode string
+		switch s.TimeInForce {
+		case order.PostOnly:
+			stpMode = "EXPIRE_MAKER"
+		case order.GoodTillCancel:
+			stpMode = "EXPIRE_TAKER"
+		}
+		switch s.Type {
+		case order.Stop, order.StopLimit, order.TrailingStop, order.TrailingStopLimit:
+			var trackingDistance string
+			if s.Type|order.TrailingStop == order.TrailingStop {
+				trackingDistance = strconv.FormatFloat(s.TrackingValue, 'f', -1, 64)
+				if s.TrackingMode == order.Percentage {
+					trackingDistance += "%"
+				}
+			}
+
+			sOrder, err := e.CreateSmartOrder(ctx, &SmartOrderRequest{
+				Symbol:         s.Pair,
+				Type:           OrderType(s.Type),
+				Side:           s.Side,
+				AccountType:    AccountType(s.AssetType),
+				Price:          s.Price,
+				StopPrice:      s.TriggerPrice,
+				Quantity:       s.Amount,
+				Amount:         s.QuoteAmount,
+				ClientOrderID:  s.ClientOrderID,
+				TimeInForce:    TimeInForce(s.TimeInForce),
+				TrailingOffset: trackingDistance,
+			})
+			if err != nil {
+				return nil, err
+			}
+			return s.DeriveSubmitResponse(sOrder.ID)
+		case order.Market, order.Limit, order.LimitMaker:
+			response, err := e.PlaceOrder(ctx, &PlaceOrderRequest{
```

Comment:
🚧 Change request:
I think we're missing `AccountType` here?
```suggestion
			response, err := e.PlaceOrder(ctx, &PlaceOrderRequest{
				AccountType: AccountType(s.AssetType),
```